auslib/db.py

```diff
@@ -1096,6 +1139,11 @@ def addLocaleToRelease(self, name, product, platform, locale, data, old_data_ver
             raise ValueError("Release blob contains forbidden domain.")
         where = [self.name == name]
         what = dict(data=releaseBlob.getJSON())
+
+        product = self.select(where=where, columns=[self.product], transaction=transaction)[0]["product"]
```

Comment:
updateRelease() does the permission checks early, and this does them late. Consistency is good, unless there is some reason not to; slight preference for early.